strapi/strapi
# strapi-plugin-documentation: settings.json can't override "components"

strapi-plugin-documentation: settings.json can't override property "components"



### What does it do?

Added an empty ` "components" : {},` to the default [strapi/packages/strapi-plugin-documentation/config/settings.json](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-documentation/config/settings.json) file

### Why is it needed?

Inside the documentation plugin, at the beginning of the `services/utils/Documentation.js` file [defaultSettingsKeys](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-documentation/services/Documentation.js#L19) are initialised without a `components` key. This causes the code to strip any user defined `components` on [line 510](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-documentation/services/Documentation.js#L510).

### How to test it?

Clone the [default settings.json](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-documentation/config/settings.json) to the sample app at `extensions/documentation/config/settings.json`. 

Modify the settings.json to add the extra security schemes:

```
... 
  "security": [{
    "accessTokenHeaderAuth": []
  }],
  "components": {
    ...
    "securitySchemes": {
        ...
        "accessTokenHeaderAuth": {
          "type": "apiKey",
          "name": "x-access-token",
          "in": "header"
        }
    }
 }
... 

```

Restart strapi develop and open up the documentation. Clicking on authorise should now show the new auth options. 

### Related issue(s)/PR(s)

fixes #8629
